internal/attacktechniques/k8s/credential-access/dump-secrets/main.go

```diff
@@ -0,0 +1,104 @@
+package kubernetes
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"github.com/datadog/stratus-red-team/internal/providers"
+	"github.com/datadog/stratus-red-team/pkg/stratus"
+	"github.com/datadog/stratus-red-team/pkg/stratus/mitreattack"
+	"log"
+	"strconv"
+)
+
+func init() {
+	const codeBlock = "```"
+	stratus.GetRegistry().RegisterAttackTechnique(&stratus.AttackTechnique{
+		ID:                 "k8s.credential-access.dump-secrets",
+		FriendlyName:       "Dump All Secrets",
+		Platform:           stratus.Kubernetes,
+		IsIdempotent:       true,
+		MitreAttackTactics: []mitreattack.Tactic{mitreattack.CredentialAccess},
+		Description: `
+Dumps all Secrets from a Kubernetes cluster. 
+This allow an attacker with the right permissions to trivially access all secrets in the cluster.
+
+Warm-up: None
+
+Detonation: 
+
+- Dump secrets using the **LIST /api/v1/secrets** API
+- This returns all secrets in the K8s clusters, no matter their namespace
+
+References:
+
+- https://darkbit.io/blog/the-power-of-kubernetes-rbac-list
+`,
+		Detection: `
+Using Kubernetes API server audit logs. In particular, look for **list secrets** requests that are performed
+for a specific namespace.
+
+Sample event (shortened):
+
+` + codeBlock + `json
+{
+  "apiVersion": "audit.k8s.io/v1",
+  "stage": "ResponseComplete",
+  "kind": "Event",
+  "level": "Metadata",
+  "requestURI": "/api/v1/secrets?limit=500",
+  "attributes": {
+    "objectRef": {
+      "resource": "secrets",
+      "apiVersion": "v1"
+    },
+    "http": {
+      "url_details": {
+        "path": "/api/v1/secrets",
+        "queryString": {
+          "limit": "500"
+        }
+      },
+      "method": "list"
+    }
+  }
+}
+` + codeBlock + `
+
+Some built-in Kubernetes components might need to be excluded from such a detection:
+
+- namespace-controller
+- kube-state-metrics
+- apiserver
+`,
+		Detonate: detonate,
+	})
+}
+
+func detonate(map[string]string) error {
+	client := providers.K8s().GetClient()
+
+	log.Println("Attempting to dump secrets in all namespaces")
+	result := client.CoreV1().RESTClient().Get().Resource("secrets").Do(context.Background())
```

Comment:
Could this use the `core/v1/secrets` api to make this cleaner and not require marshaling, etc?  Passing `""` as the namespace should list secrets in all namespaces.

```suggestion
	result := client.CoreV1().Secrets("").List(context.Background())
```

Reply:
thanks! I did not know that (which is why I used this API)

Reply:
fixed in [3574e63](https://github.com/DataDog/stratus-red-team/pull/106/commits/3574e639797fd1ef4e7c2163c2882f76f794df00), thanks for the suggestion